File: free_tasks/ft_js/longest_common_pref.js
class Solution {
    static longestCommonPrefix(strs) {
        let res = "";
        for (let i = 0; i < strs[0].length; i++) {
            for (let s of strs) {
                if (i === s.length || s[i] !== strs[0][i]) {
                    return res;
                }
            }
            res += strs[0][i];
        }
        return res;
    }
}

const strs = ["wliwer", "flight", "fliw", "flinger", "fliod", "flicker"];
console.log(Solution.longestCommonPrefix(strs));

File: free_tasks/ft_py/longest_common_pref.py
class Solution:
    strs = ["wliwer", "flight", "fliw","flinger","fliod","flicker"]
    def longestCommonPrefix(strs) -> str:
        res = ""
        for i in range(len(strs[0])):
            for s in strs:
                if i == len(s) or s[i] != strs[0][i]:
                    return res
            res += strs[0][i]    
        return res
    print(longestCommonPrefix(strs))